main.py

```diff
@@ -154,34 +155,32 @@ def forward(self, inp, **kwargs):
     torch.cuda.empty_cache()
 
     forward_args = {k: cache[k] for k in forward_arg_names}
+    assert cache["i"] == sum(len(inp_tensor) for inp_tensor in inps), "internal error: found empty rows in inps"
     return inps, forward_args
 
 
 @torch.no_grad()
-def quantize_aq(
-    model: PreTrainedModel, dataloader: Iterable, args: Namespace, val_dataloader: Optional[Iterable] = None
-):
+def quantize_aq(model: PreTrainedModel, data: Sequence, val_data: Optional[Sequence], args: Namespace):
     assert not torch.backends.cuda.matmul.allow_tf32
     print("\nStarting AQ quantization ...")
-    inps, forward_args = get_inps(model, dataloader, args)
+    inps, forward_args = get_inps(model, data, args.model_seqlen, args.devices, args.offload_activations)
     outs = [torch.zeros_like(inp_tensor, pin_memory=inp_tensor.is_pinned()) for inp_tensor in inps]
 
-    if val_dataloader:
+    if val_data:
         run_validation = True
-        val_inps, _ = get_inps(model, val_dataloader, args)
+        val_inps, _ = get_inps(model, val_data, args.model_seqlen, args.devices, args.offload_activations)
         val_outs = [torch.zeros_like(inp_tensor, pin_memory=inp_tensor.is_pinned()) for inp_tensor in val_inps]
     else:
         run_validation = False
         val_inps, val_outs = None, None
 
-    num_codebooks = args.num_codebooks
     use_cache = model.config.use_cache
     num_codebooks = args.num_codebooks
     model.config.use_cache = False
 
     quantizers = {}
     overall_bits = 0
-    model_number_of_params = 0
+    number_of_quantized_params = 0
```

Comment:
Rationale: this is the number of parameters quantized by this run, which is not the same as all model parameters.